pkg/deviceshifu/deviceshifubase/pushtelemetry.go

```diff
@@ -109,6 +111,37 @@ func pushToShifuTelemetryCollectionService(message *http.Response, request *v1al
 	return nil
 }
 
+func getPasswordFromSecret(c *rest.RESTClient, name, ns string) (string, error) {
+	secret := v1.Secret{}
+	secretPath := fmt.Sprintf("/api/v1/namespaces/%s/%s/%s", ns, "secrets", name)
```

Comment:
why are we using two different methods to retrieve secret? in `utils/k8s.go` we use `"k8s.io/client-go/kubernetes"`

Reply:
`client-go/kubernetes` wraps a layer on top of `client-go/rest`, which is more convenient to use. Therefore the former is used in `utils/k8s.go`.

And there is a ready-made `rest.RESTClient` in the `DeviceShifuBase` structure, [here](https://github.com/FFFFFaraway/shifu/blob/60da51054a230fe6ae47436c0cceb96730484cdc/pkg/deviceshifu/deviceshifubase/deviceshifubase.go#L21), so it has not been changed.